Sources/SourceControl/GitRepository.swift

```diff
@@ -752,7 +752,11 @@ public final class GitRepository: Repository, WorkingCheckout {
         guard let firstLine = ByteString(split[0]).validDescription else {
             return false
         }
-        return (try? localFileSystem.isDirectory(AbsolutePath(validating: firstLine))) == true
+        guard let objectsPath = try? AbsolutePath(validating: firstLine), localFileSystem.isDirectory(objectsPath) else {
+            return false
+        }
+        let repositoryPath = objectsPath.parentDirectory
+        return expected == repositoryPath
```

Comment:
this is the key change. isAlternateObjectStoreValid now also checks that the "pointer" the repo is valid not just exists. before you could easily run into a situation where the checkout would point to a valid path but outside the source tree and not the one we would run "update" on